From 01edc09aaafef97caa032409f20ff4534e49ace5 Mon Sep 17 00:00:00 2001
From: Ben Greear <greearb@candelatech.com>
Date: Wed, 25 Jan 2017 13:49:02 -0800
Subject: [PATCH 31/37] netdev: Don't spam about 'virtual device tried to queue
 packet"

We see log floods when using pktgen to drive an 802.1q vlan,
so just ensure this message is never seen more than once per
boot.

Signed-off-by: Ben Greear <greearb@candelatech.com>
---
 net/core/dev.c | 14 ++++++++++++--
 1 file changed, 12 insertions(+), 2 deletions(-)

diff --git a/net/core/dev.c b/net/core/dev.c
index ff3ba72..e2d3d4a 100644
--- a/net/core/dev.c
+++ b/net/core/dev.c
@@ -3564,8 +3564,18 @@ static int __dev_queue_xmit(struct sk_buff *skb, void *accel_priv,
 				}
 			}
 			HARD_TX_UNLOCK(dev, txq);
-			net_crit_ratelimited("Virtual device %s asks to queue packet!\n",
-					     dev->name);
+			{
+				/* This spams when using pktgen on 802.1q vlans.  Doesn't seem
+				 * to actually cause any harm, so make sure this message is not
+				 * repeated more than once. --Ben
+				 */
+				static int done_one = 0;
+				if (!done_one) {
+					net_crit_ratelimited("Virtual device %s asks to queue packet!\n",
+							     dev->name);
+					done_one = 1;
+				}
+			}
 		} else {
 			/* Recursion is detected! It is possible,
 			 * unfortunately
-- 
2.4.11

